src/gas_saved.py

```diff
@@ -0,0 +1,48 @@
+import os
+
+import click
+import pandas as pd
+from dotenv import load_dotenv
+from web3 import Web3
+
+from src.db.pg_client import pg_engine
+
+
+@click.command()
+@click.option(
+    "--batch_tx_hash",
+    help="The transaction hash of the batch you want to see the % gas saved.",
+    type=str
+)
+def main(batch_tx_hash):
+    db_engine = pg_engine()
+    GAS_QUOTES_QUERY = f"""
+    SELECT orders.id::bytea, order_quotes.gas_amount, order_quotes.gas_price
+    FROM
+        solver_competitions,
+        jsonb_to_recordset(solver_competitions.json->'solutions'->-1->'orders') AS orders(id text)
+    LEFT JOIN order_quotes ON order_quotes.order_uid = ('\\' || LTRIM(orders.id::text, '0'))::bytea
+    WHERE
+      solver_competitions.tx_hash = '\\{batch_tx_hash[1:]}'
```

Comment:
Do you actually know how smart postgres is?
Does your query have the same performance than this one?
```
with competition as(
SELECT        jsonb_to_recordset(solver_competitions.json->'solutions'->-1->'orders') AS orders(id text)
    FROM
        solver_competitions
WHERE
      solver_competitions.tx_hash = '\\{batch_tx_hash[1:]}'
        ),
select orders.id::bytea, order_quotes.gas_amount, order_quotes.gas_price from competition LEFT JOIN order_quotes ON order_quotes.order_uid = ('\\' || LTRIM(orders.id::text, '0'))::bytea